lib/ansible/modules/cloud/azure/azure_rm_autoscale.py

```diff
@@ -0,0 +1,620 @@
+                                                               if p.get('recurrence_frequency') else None)) for p in self.profiles or []]
+
+            notifications = [AutoscaleNotification(email=EmailNotification(**n),
+                                                   webhooks=[WebhookNotification(service_uri=w) for w in n.get('webhooks') or []])
+                             for n in self.notifications or []]
+
+            if not results:
+                # create new
+                changed = True
+            else:
+                # check changed
+                resource_name = results.autoscale_setting_resource_name or self.name
+                update_tags, tags = self.update_tags(results.tags)
+                if update_tags:
+                    changed = True
+                    self.tags = tags
+                if self.target != results.target_resource_uri:
+                    changed = True
+                if self.enabled != results.enabled:
+                    changed = True
+                profile_result_set = set([str(profile_to_dict(p)) for p in results.profiles or []])
+                if profile_result_set != set([str(profile_to_dict(p)) for p in profiles]):
+                    changed = True
+                notification_result_set = set([str(notification_to_dict(n)) for n in results.notifications or []])
+                if notification_result_set != set([str(notification_to_dict(n)) for n in notifications]):
+                    changed = True
+            if changed:
+                # construct the instance will be send to create_or_update api
+                results = AutoscaleSettingResource(location=self.location,
+                                                   tags=self.tags,
+                                                   profiles=profiles,
+                                                   notifications=notifications,
+                                                   enabled=self.enabled,
+                                                   autoscale_setting_resource_name=resource_name,
+                                                   target_resource_uri=self.target)
+                if not self.check_mode:
+                    results = self.create_or_update_auto_scale(results)
+                # results should be the dict of the instance
+        self.results = auto_scale_to_dict(results)
+        self.results['changed'] = changed
+        return self.results
+
+    def get_auto_scale(self):
+        try:
+            return self.monitor_client.autoscale_settings.get(self.resource_group, self.name)
+        except Exception as exc:
+            self.log('Error: failed to get auto scale settings {0} - {1}'.format(self.name, str(exc)))
+            return None
+
+    def create_or_update_auto_scale(self, param):
+        try:
+            return self.monitor_client.autoscale_settings.create_or_update(self.resource_group, self.name, param)
```

Comment:
why not to add auto_scale_to_doct in this method. caller doesn't need to care about refine output and how to map then.